lib/Parse/Lexer.cpp

```diff
@@ -1477,6 +1477,29 @@ void Lexer::lexImpl() {
       } else {
         diagnose(CurPtr-1, diag::lex_invalid_character)
           .fixItReplaceChars(getSourceLoc(CurPtr-1), getSourceLoc(tmp), " ");
+
+        StringRef confusedChar;
+        if (codepoint >= 0x80) {
+          llvm::SmallString<64> buffer;
+          EncodeToUTF8(codepoint, buffer);
+          confusedChar = buffer.str();
+        } else {
+          confusedChar = StringRef(std::string(1, (char)codepoint));
```

Comment:
Agreed, but `EncodeToUTF8` asserts that codepoint has to be greater than `0x80`, and I wasn't sure that I should change the contract of the function.